xla/hlo/analysis/hlo_replication_analysis.cc

```diff
@@ -78,11 +156,20 @@ HloReplicationAnalysis::DetermineHloInstructionIsReplicated(
         return HloReplication::ReplicatedOnAllDevices();
       }
       if (support_partial_replication) {
-        std::vector<absl::Span<const int64_t>> device_sets;
+        const int64_t num_partitions =
+            hlo->GetModule()->config().num_partitions();
+        std::vector<std::vector<std::vector<int64_t>>> groups_for_replicas(
+            num_partitions);
```

Comment:
I think it's slightly more elegant to create a doubly-nested vector, std::vector<std::vector<int64_t>>, fill it out, then call `std::fill` on it. You can name it `groups`